src/plugins/data/public/search/session/session_service.ts

```diff
@@ -263,7 +289,33 @@ export class SessionService {
 
     // if we are still interested in this result
     if (this.getSessionId() === sessionId) {
-      this.state.transitions.store();
+      this.state.transitions.store(searchSessionSavedObject);
+    }
+  }
+
+  /**
+   * Change user-facing name of a current session
+   * Doesn't throw in case of API error but presents a notification toast instead
+   * @param newName - new session name
+   */
+  public async renameCurrentSession(newName: string) {
+    const sessionId = this.getSessionId();
+    if (sessionId && this.state.get().isStored) {
+      let renamed = false;
+      try {
+        await this.sessionsClient.rename(sessionId, newName);
+        renamed = true;
+      } catch (e) {
+        this.toastService?.addError(e, {
```

Comment:
This will never be called as sessionsClient.rename already swallows the error and shows toast notifications.